yt/testing.py

```diff
@@ -154,11 +157,11 @@ def integrate_kernel(
 
 
 def distancematrix(
-    pos3_i0: np.ndarray,
-    pos3_i1: np.ndarray,
+    pos3_i0: NDArray[np.floating],
+    pos3_i1: NDArray[np.floating],
     periodic: tuple[bool, bool, bool] = (True,) * 3,
-    periods: np.ndarray = _zeroperiods,
-) -> np.ndarray:
```

Comment:
actually here the output dtype is bound to be the same as `pos3_i3`, so, instead of `np.floating`, we should use `_FloatingT = TypeVar("_FloatingT", bound=np.floating)` here